### ISSUE TITLE
Vault upstream authority unit tests fail when VAULT_ADDR is set in the user's environment

### ISSUE THREAD

**Issue by rturner3** [spiffe/spire#1713]
* **Version**: 89bccc5f
* **Platform**: `Darwin <CENSORED> 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64`
* **Subsystem**: pkg/server/plugin/upstreamauthority/vault/vault_test.go

Reproduce steps:
```bash
$ cd /path/to/spire
$ export VAULT_ADDR=https://somevaulthost:443
$ make build test
```

Unit tests should ideally not be affected by environment variables set in the user's shell.

**Comment by azdagron** [comment]
This should be an easy fix:
1. Evaluate what environment variables are picked up by the vault client library used by SPIRE
2. Add an `init()` function to the vault_test.go file that unset's those environment variables at init-time.

**Comment by azdagron** [comment]
This was fixed by #1721 .
